client.go

```diff
@@ -268,6 +276,23 @@ func CompressionCodec(codec string) Option {
 	}
 }
 
+// ZooKeeperDialer will return an option to pass the given dialer function
+// into the ZooKeeper client Connect() call, which allows for customizing
+// network connections.
+func ZooKeeperDialer(dialer gzk.Dialer) Option {
+	return func(c *client) {
+		c.zkDialer = dialer
+	}
+}
+
+// RegionDialer will return an option that uses the specified Dialer for
+// connecting to region servers. This allows for connecting through proxies.
+func RegionDialer(dialer region.Dialer) Option {
```

Comment:
As suggested above, this should take in a `func(ctx context.Context, network, addr string) (net.Conn, error)`.
I don't think we have to give a name to the type.